File: mycollab-web/src/main/java/com/esofthead/mycollab/shell/view/MainViewController.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package com.esofthead.mycollab.shell.view;

import com.esofthead.mycollab.module.crm.view.CrmModulePresenter;
import com.esofthead.mycollab.module.crm.view.CrmModuleScreenData;
import com.esofthead.mycollab.module.file.view.FileModulePresenter;
import com.esofthead.mycollab.module.file.view.FileModuleScreenData;
import com.esofthead.mycollab.module.project.view.ProjectModulePresenter;
import com.esofthead.mycollab.module.project.view.ProjectModuleScreenData;
import com.esofthead.mycollab.module.user.accountsettings.view.AccountModulePresenter;
import com.esofthead.mycollab.module.user.accountsettings.view.AccountModuleScreenData;
import com.esofthead.mycollab.shell.events.ShellEvent;
import com.esofthead.mycollab.vaadin.events.ApplicationEvent;
import com.esofthead.mycollab.vaadin.events.ApplicationEventListener;
import com.esofthead.mycollab.vaadin.events.EventBus;
import com.esofthead.mycollab.vaadin.mvp.IController;
import com.esofthead.mycollab.vaadin.mvp.PresenterResolver;

/**
 * 
 * @author haiphucnguyen
 */
public class MainViewController implements IController {
	private static final long serialVersionUID = 1L;
	private MainView container;

	public MainViewController(MainView view) {
		this.container = view;
		bind();

	}

	private void bind() {
		EventBus.getInstance().addListener(
				new ApplicationEventListener<ShellEvent.GotoCrmModule>() {
					private static final long serialVersionUID = 1L;

					@Override
					public Class<? extends ApplicationEvent> getEventType() {
						return ShellEvent.GotoCrmModule.class;
					}

					@Override
					public void handle(ShellEvent.GotoCrmModule event) {
						CrmModulePresenter crmModulePresenter = PresenterResolver
								.getPresenter(CrmModulePresenter.class);
						CrmModuleScreenData.GotoModule screenData = new CrmModuleScreenData.GotoModule(
								(String[]) event.getData());
						crmModulePresenter.go(container, screenData);
					}
				});

		EventBus.getInstance().addListener(
				new ApplicationEventListener<ShellEvent.GotoProjectModule>() {
					private static final long serialVersionUID = 1L;

					@Override
					public Class<? extends ApplicationEvent> getEventType() {
						return ShellEvent.GotoProjectModule.class;
					}

					@Override
					public void handle(ShellEvent.GotoProjectModule event) {
						ProjectModulePresenter prjPresenter = PresenterResolver
								.getPresenter(ProjectModulePresenter.class);
						ProjectModuleScreenData.GotoModule screenData = new ProjectModuleScreenData.GotoModule(
								(String[]) event.getData());
						prjPresenter.go(container, screenData);
					}
				});

		EventBus.getInstance()
				.addListener(
						new ApplicationEventListener<ShellEvent.GotoUserAccountModule>() {
							private static final long serialVersionUID = 1L;

							@Override
							public Class<? extends ApplicationEvent> getEventType() {
								return ShellEvent.GotoUserAccountModule.class;
							}

							@Override
							public void handle(
									ShellEvent.GotoUserAccountModule event) {
								AccountModulePresenter presenter = PresenterResolver
										.getPresenter(AccountModulePresenter.class);
								presenter.go(container,
										new AccountModuleScreenData.GotoModule(
												(String[]) event.getData()));
							}
						});

		EventBus.getInstance().addListener(
				new ApplicationEventListener<ShellEvent.GotoFileModule>() {
					private static final long serialVersionUID = 1L;

					@Override
					public Class<? extends ApplicationEvent> getEventType() {
						return ShellEvent.GotoFileModule.class;
					}

					@Override
					public void handle(ShellEvent.GotoFileModule event) {
						FileModulePresenter fileModulePresenter = PresenterResolver
								.getPresenter(FileModulePresenter.class);
						FileModuleScreenData.GotoModule screenData = new FileModuleScreenData.GotoModule(
								(String[]) event.getData());
						fileModulePresenter.go(container, screenData);
					}
				});
	}
}

File: mycollab-web/src/main/scala/com/esofthead/mycollab/shell/view/MainViewController.scala
package com.esofthead.mycollab.shell.view

import com.esofthead.mycollab.core.utils.ScalaUtils
import com.esofthead.mycollab.eventmanager.ApplicationEventListener
import com.esofthead.mycollab.module.crm.view.{CrmModulePresenter, CrmModuleScreenData}
import com.esofthead.mycollab.module.file.view.{FileModuleScreenData, IFileModulePresenter}
import com.esofthead.mycollab.module.project.view.ProjectModulePresenter
import com.esofthead.mycollab.module.project.view.parameters.ProjectModuleScreenData
import com.esofthead.mycollab.module.user.accountsettings.view.AccountModulePresenter
import com.esofthead.mycollab.premium.module.user.accountsettings.view.AccountModuleScreenData
import com.esofthead.mycollab.shell.events.ShellEvent
import com.esofthead.mycollab.vaadin.mvp.{AbstractController, PresenterResolver}
import com.google.common.eventbus.Subscribe

/**
  * @author MyCollab Ltd
  * @since 5.0.9
  */
class MainViewController(val container: MainView) extends AbstractController {
  bind()

  private def bind(): Unit = {
    this.register(new ApplicationEventListener[ShellEvent.GotoCrmModule]() {
      @Subscribe def handle(event: ShellEvent.GotoCrmModule) {
        val crmModulePresenter = PresenterResolver.getPresenter(classOf[CrmModulePresenter])
        val screenData = new CrmModuleScreenData.GotoModule(ScalaUtils.stringConvertSeqToArray(event.getData))
        crmModulePresenter.go(container, screenData)
      }
    })
    this.register(new ApplicationEventListener[ShellEvent.GotoProjectModule]() {
      @Subscribe def handle(event: ShellEvent.GotoProjectModule) {
        val prjPresenter = PresenterResolver.getPresenter(classOf[ProjectModulePresenter])
        val screenData = new ProjectModuleScreenData.GotoModule(ScalaUtils.stringConvertSeqToArray(event.getData))
        prjPresenter.go(container, screenData)
      }
    })
    this.register(new ApplicationEventListener[ShellEvent.GotoUserAccountModule]() {
      @Subscribe def handle(event: ShellEvent.GotoUserAccountModule) {
        val presenter = PresenterResolver.getPresenter(classOf[AccountModulePresenter])
        presenter.go(container, new AccountModuleScreenData.GotoModule(ScalaUtils.stringConvertSeqToArray(event.getData)))
      }
    })
    this.register(new ApplicationEventListener[ShellEvent.GotoFileModule]() {
      @Subscribe def handle(event: ShellEvent.GotoFileModule) {
        val fileModulePresenter = PresenterResolver.getPresenter(classOf[IFileModulePresenter])
        val screenData = new FileModuleScreenData.GotoModule(ScalaUtils.stringConvertSeqToArray(event.getData))
        fileModulePresenter.go(container, screenData)
      }
    })
  }
}
